cmd/tsnetd/main.go

```diff
@@ -0,0 +1,158 @@
+// Server tsnetd "tailscale inetd" is a TCP proxy that can listen on tsnet addresses.
+//
+// My main motivation for this is to run multiple services on the same host, but give
+// them memorable names and use cononical ports.
+//
+// Usage:
+//
+//	tsnetd tailscale-host-1:tailscale-port-1:target-host:target-port ...
+//
+// For example:
+//
+//	tsnetd cameras:http:localhost:8001 cameras:rtsp:localhost:rtsp phone:sip:localhost:sip
+//
+// This will register two nodes on your tailnet, "cameras" and "phone", using the auth key in
+// the environment variable TS_AUTHKEY. On cameras it will forward port 80 to localhost:8001
+// and port 554 (rtsp) to localhost:554, and on phone it will forward port 5060 (sip) to
+// localhost:5060.
+//
+// You can get the same effect if you:
+//  1. Run multiple tailscaleds in separate network namespaces or containers, but that can get complicated.
+//  2. Use the caddy-tailscale extension, but that's HTTP only.
+//  2. Use an HTTP proxy & vitual hosts, but now you have to set your own DNS. Also HTTP (or TLS) only.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"tailscale.com/tsnet"
+)
+
+var logf = func(string, ...any) {}
+
+func proxy(s *tsnet.Server, a net.Conn, dst string) {
```

Comment:
can we do this using serve now?

Reply:
ah so you're basically running multiple tsnet instances here, so you could still use serve on a per tsnet.Server basis.